### DISCUSSION TITLE
Messaging between frontend and backend

### DISCUSSION THREAD

**Discussion by fouadhallal** [fastapi/fastapi#11485] [Questions] [answered]
Hi,  I am new to Fastapi. I am going through Git repo of full stack fastapi template and trying to understand the different pieces.  My understanding is the template assumes that both front end and back end are running locally? In the case they are not, what messaging infrastructure do you recommend to communicate between the web application and the backend service hosted in the cloud? Any code examples?  Any information would help.

Thank you

**Comment by YuriiMotov** [comment, marked as answer]
Hello!

Front-end part of web application communicates with back-end part via HTTP protocol.
They can be hosted on different servers. Front-end part should know the origin (combination of protocol, domain and port) of back-end part. At the same time, back-end part should include the origin of front-end into the list of allowed origins to avoid issues with CORS (https://fastapi.tiangolo.com/tutorial/cors/).

I'm not quite familiar with 'full stack fastapi template', but I think it can be deployed on different servers.
As I understand, to make it work you should:

**1. Adjust ENV-variables:**

In the `.env` file wich placed in the root directory of the project:

- `BACKEND_CORS_ORIGINS` - url of your front-end (e.g. "https://myfrontend.com:3000")
- database server connection parameters (`POSTGRES_SERVER`, `POSTGRES_PORT`, etc..)

In the `frontend/src/.env` file:

- `VITE_API_URL` - url of your back-end (e.g. "https://example.com:8000")

**2. Deploy front-end, back-end and database Docker containers on servers**

You don't need to use docker-compose here. You should run Docker containers separately

**Reply by fouadhallal** [reply]
Thank you for the explanation
